Lib/ufo2ft/featureWriters/kernFeatureWriter2.py

```diff
@@ -208,7 +224,35 @@ def getKerningClasses(cls, font, feaFile=None, glyphSet=None):
         return side1Classes, side2Classes
 
     @staticmethod
-    def getKerningPairs(font, side1Classes, side2Classes, glyphSet=None):
+    def getKerningPairs(font, side1Classes, side2Classes, glyphSet=None, options=None):
+        if isinstance(font, DesignSpaceDocument):
+            # Reuse the newer kern writers variable kerning extractor. Repack
+            # some arguments and the return type for this.
+            side1ClassesRaw: Mapping[str, tuple[str, ...]] = {
+                group_name: tuple(
+                    glyph
+                    for glyphs in glyph_defs.glyphSet()
+                    for glyph in glyphs.glyphSet()
+                )
+                for group_name, glyph_defs in side1Classes.items()
+            }
+            side2ClassesRaw: Mapping[str, tuple[str, ...]] = {
+                group_name: tuple(
+                    glyph
+                    for glyphs in glyph_defs.glyphSet()
+                    for glyph in glyphs.glyphSet()
+                )
+                for group_name, glyph_defs in side2Classes.items()
+            }
+            pairs = NewKernFeatureWriter.getVariableKerningPairs(
+                font,
+                side1ClassesRaw,
+                side2ClassesRaw,
+                glyphSet or {},
+                options or KernFeatureWriter.options,
```

Comment:
options parameter is supposed to be a SimpleNamespace, but KernFeatureWriter.options class attribute is actually a dict, I suggest you convert it into a namespace before passing it onwards